tensorflowjs/prePost.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const tf = require('@tensorflow/tfjs');
+const fs = require('fs');
+const {Image, createCanvas} = require('canvas');
+
+
+class PrePost {
+
+    preprocess(request) {
+        throw {name: "NotImplementedError", message: "This function needs to be implemented."};
+    }
+
+    postprocess(request) {
+        throw {name: "NotImplementedError", message: "This function needs to be implemented."}
+    }
+
+    errorPostProcess(request, errorMessage) {
+        throw {name: "NotImplementedError", message: "This function needs to be implemented."}
+    }
+}
+
+class ObjectDetectorHandler extends PrePost {
+
+    preprocess(request) {
+    
+        request.processedInputs['image'] = request.rawInputs['image'].map(element => {
+            let base64Encoded = Buffer.from(element).toString('base64');
```

Comment:
The input image should already be in base64 format when sent to the web service. If you are finding that it is not in that format at this point in the flow, then Node is decoding the base64 string for you.